src/overlays/actors/ovl_Boss_Va/z_boss_va.c

```diff
@@ -173,157 +319,3695 @@ static ColliderQuadInit D_8095BFF4 = {
+        Math_SmoothStepToS(&this->unk_1F0, this->skelAnime.jointTable[7].z, 1, 0xEA6, 0);
+        Math_SmoothStepToF(&this->skelAnime.playSpeed, 1.0f, 1.0f, 0.05f, 0.0f);
+        this->burst = false;
+    }
+
+    if (this->burst && (this->burst != 2)) { // burst can never be 2
+        if (this->timer2 >= 16) {
+            if (this->timer2 == 18) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_BALINADE_THUNDER);
+            }
+
+            BossVa_Spark(globalCtx, this, 2, 110, 15.0f, 15.0f, SPARK_BLAST, 5.0f, true);
+            BossVa_Spark(globalCtx, this, 2, 110, 15.0f, 15.0f, SPARK_BLAST, 6.0f, true);
+            BossVa_Spark(globalCtx, this, 2, 110, 15.0f, 15.0f, SPARK_BLAST, 7.0f, true);
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderLightning.base);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderLightning.base);
+        } else {
+            BossVa_Spark(globalCtx, this, 2, 50, 15.0f, 0.0f, SPARK_BODY, (this->timer2 >> 1) + 1, true);
+            if (this->timer2 == 14) {
+                BossVa_SetSparkEnv(globalCtx);
+            }
+            if (this->timer2 == 4) {
+                Vec3f sp48 = this->zapHeadPos;
+
+                BossVa_SpawnZapperCharge(globalCtx, sVaEffects, this, &sp48, &this->headRot, 100, 0);
+            }
+        }
+
+        this->timer2++;
+        if (this->timer2 >= 24) {
+            this->burst = false;
+        }
+    }
+}
+
+void BossVa_SetupZapperHold(BossVa* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelAnime, &gBarinadeZapperDamage2Anim, 0.0f, 0.0f, Animation_GetLastFrame(&gBarinadeZapperDamage2Anim),
+                     ANIMMODE_ONCE_INTERP, -6.0f);
+    this->burst = false;
+    BossVa_SetupAction(this, BossVa_ZapperHold);
+}
+
+void BossVa_ZapperHold(BossVa* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    BossVa_AttachToBody(this);
+    Math_SmoothStepToS(&this->unk_1E6, 0, 1, 0x1770, 0);
+    Math_SmoothStepToS(&this->unk_1E4, 0, 1, 0x1770, 0);
+    Math_SmoothStepToS(&this->unk_1EC, 0, 1, 0x1770, 0);
+    Math_SmoothStepToS(&this->unk_1EA, 0, 1, 0x1770, 0);
+    Math_SmoothStepToS(&this->unk_1F2, this->actor.shape.rot.y - 0x4000, 1, 0x2710, 0);
+    Math_SmoothStepToS(&this->unk_1F0, this->skelAnime.jointTable[7].z - 0x1388, 1, 0x1770, 0);
+    if (BODY->actor.speedXZ == 0.0f) {
+        BossVa_SetupZapperAttack(this, globalCtx);
+    }
+}
+
+void BossVa_SetupBariIntro(BossVa* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelAnime, &gBarinadeBariAnim, 1.0f, 0.0f, Animation_GetLastFrame(&gBarinadeBariAnim),
+                     ANIMMODE_LOOP, 0.0f);
+    this->unk_1A0 = 60.0f;
+    this->unk_1A4 = Rand_ZeroOne() * 360.0f;
+    this->timer2 = 64;
+    this->unk_1F0 = 120;
+    this->unk_1A8 = 0.0f;
+    this->actor.world.pos.x = sInitPosOffsets[this->actor.params + 10].x + this->actor.home.pos.x;
```

Comment:
```suggestion
    this->actor.world.pos.x = sInitPosOffsets[this->actor.params + (BOSSVA_BARI_LOWER_5 - BOSSVA_BARI_UPPER_1 + 1)].x + this->actor.home.pos.x;
```
Not sure, a `BOSSVA_BARI_COUNT` define may come in handy too
Or just keep the magic 10 since the `sInitPosOffsets` is kind of magic indexing anyway